Explorer/Assets/DCL/Landscape/Systems/LandscapeMiscCullingSystem.cs

```diff
@@ -0,0 +1,232 @@
+﻿using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using Arch.SystemGroups.DefaultSystemGroups;
+using DCL.CharacterCamera;
+using DCL.Diagnostics;
+using DCL.Landscape.Jobs;
+using DCL.Landscape.Settings;
+using ECS.Abstract;
+using System.Collections.Generic;
+using System.Linq;
+using Unity.Collections;
+using Unity.Jobs;
+using Unity.Mathematics;
+using UnityEngine;
+using UnityEngine.Profiling;
+using Vector3 = UnityEngine.Vector3;
+
+namespace DCL.Landscape.Systems
+{
+    /// <summary>
+    ///     This system updates every terrain visibility (terrain and trees) based on the camera position and distance settings
+    ///     We schedule a parallel job to calculate the frustum and distance culling and then we use the job results to set-up the correct flags
+    /// </summary>
+    [LogCategory(ReportCategory.LANDSCAPE)]
+    [UpdateInGroup(typeof(PresentationSystemGroup))]
+    public partial class LandscapeMiscCullingSystem : BaseUnityLoopSystem
+    {
+        private readonly LandscapeData landscapeData;
+        private readonly TerrainGenerator terrainGenerator;
+        private readonly List<MeshRenderer> cliffRenderers = new ();
+
+        private bool isSetupDone;
+
+        private NativeArray<float4> nativeFrustumPlanes;
+        private NativeArray<VisibleBounds> cliffsBoundaries;
+        private NativeArray<VisibleBounds> waterBoundaries;
+        private Plane[] frustumPlanes;
+        private JobHandle cliffsJobHandle;
+        private JobHandle waterJobHandle;
+        private bool cliffsUpdated;
+        private bool waterUpdated;
+        private List<MeshRenderer> waterRenderers;
+        private Vector3 cameraPosition;
+
+        private LandscapeMiscCullingSystem(World world,
+            LandscapeData landscapeData,
+            TerrainGenerator terrainGenerator) : base(world)
+        {
+            this.landscapeData = landscapeData;
+            this.terrainGenerator = terrainGenerator;
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            frustumPlanes = new Plane[6];
+            nativeFrustumPlanes = new NativeArray<float4>(6, Allocator.Persistent);
+            cliffsJobHandle = default(JobHandle);
+            waterJobHandle = default(JobHandle);
+        }
+
+        public override void Dispose()
+        {
+            base.Dispose();
+            cliffsJobHandle.Complete();
+            waterJobHandle.Complete();
+            nativeFrustumPlanes.Dispose();
+            cliffsBoundaries.Dispose();
+            waterBoundaries.Dispose();
+        }
+
+        protected override void Update(float t)
+        {
+            if (!terrainGenerator.IsTerrainGenerated()) return;
+
+            if (!isSetupDone)
+            {
+                isSetupDone = true;
+                InitializeMiscVisibility();
+                InitializeWaterVisibility();
+            }
+
+            if (!isSetupDone) return;
```

Comment:
Always false?

Reply:
you are right, this statement is useless